contracts/SortitionPool.sol

```diff
@@ -26,10 +36,46 @@ contract SortitionPool is SortitionTree, Ownable {
     _;
   }
 
-  constructor(uint256 _poolWeightDivisor) {
+  /// @notice Reverts if called while pool is unlocked.
+  modifier onlyLocked() {
+    require(isLocked, "Sortition pool unlocked");
+    _;
+  }
+
+  constructor(
+    IStaking _stakingContract,
+    IERC20WithPermit _rewardToken,
+    uint256 _poolWeightDivisor
+  ) {
+    stakingContract = _stakingContract;
+    rewardToken = _rewardToken;
     poolWeightDivisor = _poolWeightDivisor;
   }
 
+  function receiveApproval(
+    address sender,
+    uint256 amount,
+    address token,
+    bytes calldata
+  ) external override {
+    require(token == address(rewardToken), "Unsupported token");
+    rewardToken.transferFrom(sender, address(this), amount);
+    Rewards.addRewards(uint96(amount), uint32(root.sumWeight()));
+  }
+
+  function withdrawRewards(address operator) public {
+    uint32 id = getOperatorID(operator);
+    Rewards.updateOperatorRewards(id, uint32(getPoolWeight(operator)));
+    uint96 earned = Rewards.withdrawOperatorRewards(id);
+    (, address beneficiary, ) = stakingContract.rolesOf(operator);
+    rewardToken.transfer(beneficiary, uint256(earned));
+  }
+
+  function withdrawIneligible() public onlyOwner {
```

Comment:
Instead of transferring to `owner()` can we have the recipient passed as a parameter? This gives better flexibility.

Reply:
👍